src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -4,49 +4,55 @@
  */
 package io.strimzi.kafka.quotas;
 
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.Gauge;
+import com.yammer.metrics.core.MetricName;
+import org.apache.kafka.common.Cluster;
+import org.apache.kafka.common.metrics.Quota;
+import org.apache.kafka.common.security.auth.KafkaPrincipal;
+import org.apache.kafka.server.quota.ClientQuotaCallback;
+import org.apache.kafka.server.quota.ClientQuotaEntity;
+import org.apache.kafka.server.quota.ClientQuotaType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.nio.file.FileStore;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.AbstractMap;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
-import com.yammer.metrics.Metrics;
-import com.yammer.metrics.core.Gauge;
-import com.yammer.metrics.core.MetricName;
-
-import org.apache.kafka.common.Cluster;
-import org.apache.kafka.common.metrics.Quota;
-import org.apache.kafka.common.security.auth.KafkaPrincipal;
-import org.apache.kafka.server.quota.ClientQuotaCallback;
-import org.apache.kafka.server.quota.ClientQuotaEntity;
-import org.apache.kafka.server.quota.ClientQuotaType;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 /**
  * Allows configuring generic quotas for a broker independent of users and clients.
  */
 public class StaticQuotaCallback implements ClientQuotaCallback {
     private static final Logger log = LoggerFactory.getLogger(StaticQuotaCallback.class);
     private volatile Map<ClientQuotaType, Quota> quotaMap = new HashMap<>();
     private final AtomicLong storageUsed = new AtomicLong(0);
+    private final AtomicReference<Map<FileStore, Double>> storagePercents = new AtomicReference<>(Collections.emptyMap());
     private volatile List<Path> logDirs;
+    private volatile Map<FileStore, Map.Entry<Double, Double>> percentsMap;
```

Comment:
I think a trivial `Threshold` class, rather than using `Map.Entry` would make the code much clearer.